src/backend/data/feed.js

```diff
@@ -37,7 +44,34 @@ class Feed {
    * Returns a Promise.
    */
   save() {
-    addFeed(this);
+    return addFeed(this);
+  }
+
+  /**
+   * Removes current Feed + associated Posts from the databases.
+   * Returns a Promise
+   */
+  async delete() {
+    await removeFeed(this.id);
+    const posts = await getPosts(0, 0);
```

Comment:
This line and 55 could be `await Promise.all([ ...remove feed, ...get and remove posts])`

Reply:
changed